src/Client.h

```diff
@@ -250,25 +302,31 @@ class Client : public ClientBase{
 										  serverNo);
 				m_switchDataIntegrity.execute(m_pMsgRequest, m_pMsgReply);
 			}
+			
+#ifdef  USING_VMA_EXTRA_API
+			if (tmp_vma_poll_buff && !nbytes){
+				tmp_vma_poll_buff = tmp_vma_poll_buff->next;
+			}
+#endif
 		}
 
+
 		/* 6: shift to start of cycle buffer in case receiving buffer is empty and
 		 * there is no uncompleted message
 		 */
 		if (!nbytes) {
 			g_fds_array[ifd]->recv.cur_addr = g_fds_array[ifd]->recv.buf;
 			g_fds_array[ifd]->recv.cur_size = g_fds_array[ifd]->recv.max_size;
 			g_fds_array[ifd]->recv.cur_offset = 0;
-		}
-
+			}
```

Comment:
can you clean indentation changes and white line changes that are not needed for your commit and resubmit